sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -93,7 +93,14 @@ private[sql] class DataFrameImpl protected[sql](
       queryExecution.analyzed.resolve(n.name, sqlContext.analyzer.resolver).get
     }
   }
-
+ 
+  protected[sql] def numericColumns(colNames: String*): Seq[Expression] = {
```

Comment:
It is used to obtain these numeric columns we want if they are in the df. So we can just aggregate on them instead of all numeric columns.